src/components/spreadsheet/data-fetching/use-spreadsheet-equipments.ts

```diff
@@ -177,50 +156,19 @@ export const useSpreadsheetEquipments = (
 
     useEffect(() => {
         if (
-            shouldFetchEquipments &&
-            studyUuid &&
-            currentRootNetworkUuid &&
+            nodeIdsToFetch?.length &&
             currentNode?.id &&
             isNetworkModificationTreeModelUpToDate &&
             isNodeBuilt(currentNode)
         ) {
             setErrorMessage(null);
             setIsFetching(true);
-            let fetchers: Promise<unknown>[] = [];
-            let spreadsheetEquipmentsByNodes: SpreadsheetEquipmentsByNodes = {
-                nodesId: [],
-                equipmentsByNodeId: {},
-            };
-
-            nodesIdToFetch.forEach((nodeId) => {
-                const fetcherPromises = getFetcher(type)(studyUuid, nodeId as UUID, currentRootNetworkUuid, []);
-                fetchers.push(fetcherPromises);
-                fetcherPromises.then((results) => {
-                    let fetchedEquipments = results.flat();
-                    spreadsheetEquipmentsByNodes.nodesId.push(nodeId);
-                    if (formatFetchedEquipments) {
-                        fetchedEquipments = formatFetchedEquipments(fetchedEquipments);
-                        spreadsheetEquipmentsByNodes.equipmentsByNodeId[nodeId] = fetchedEquipments;
-                    }
-                });
-            });
-
-            Promise.all(fetchers).then(() => {
-                dispatch(loadEquipments(type, spreadsheetEquipmentsByNodes));
+            const promises = loadEquipmentData(nodeIdsToFetch);
```

Comment:
it's kind of weird to call loadequipmentData(nodeIdsToFetch) when both comes from the hook, we want to call only loadEquipmentData() from the hook

Reply:
yes

Reply:
outdated